components/table/SearchFilterSort.tsx

```diff
@@ -54,19 +54,21 @@ export default function SearchFilterSort({ showDownloadButton = false, apiContex
       }
 
       if (response.status === 204) {
-        console.log('No data to download');
         alert('No data available for download.');
         return;
       }
 
       const blob = await response.blob();
-
+      const contentDisposition = response.headers.get('Content-Disposition');
+      const filename = contentDisposition
+        ? contentDisposition.split('filename=')[1].replace(/['"]/g, '')
+        : 'default-filename.csv';
```

Comment:
Let's put actual file name that can be used as a `default filename` for this purpose.